### PR TITLE
Added check for MAILGUN_URL and MAILGUN_KEY

### PR DESCRIPTION
**Pull request by noisecapella**
Related to https://github.com/mitodl/micromasters/issues/3599

### CODE REVIEW

File: {{ cookiecutter.project_name }}/{{ cookiecutter.project_name }}/settings.py [hunk 1]
```diff
@@ -184,8 +184,12 @@
 EMAIL_SUPPORT = get_string('{{ cookiecutter.project_name|upper }}_SUPPORT_EMAIL', 'support@example.com')
 DEFAULT_FROM_EMAIL = get_string('{{ cookiecutter.project_name|upper }}_FROM_EMAIL', 'webmaster@localhost')
 
-MAILGUN_URL = get_string('MAILGUN_URL', 'https://api.mailgun.net/v3/micromasters.mit.edu')
+MAILGUN_URL = get_string('MAILGUN_URL', None)
+if not MAILGUN_URL:
```
Comment (rhysyngsun) [inline {{ cookiecutter.project_name }}/{{ cookiecutter.project_name }}/settings.py]: Could you add `MAILGUN_URL` to `.env.example` too?

File: {{ cookiecutter.project_name }}/{{ cookiecutter.project_name }}/settings_test.py [hunk 2]
```diff
@@ -13,6 +13,12 @@
 import semantic_version
 
 
+REQUIRED_SETTINGS = {
+    'MAILGUN_URL': 'http://fake.mailgun.url',
+    'MAILGUN_KEY': 'fake_mailgun_key',
+}
+
+
 class TestSettings(TestCase):
```
Comment (rhysyngsun) [inline {{ cookiecutter.project_name }}/{{ cookiecutter.project_name }}/settings_test.py]: Could you backport your test in MM for `MAILGUN_*` settings to here too?

File: {{ cookiecutter.project_name }}/{{ cookiecutter.project_name }}/settings.py (hunk 1 above)
Comment (noisecapella) [reply {{ cookiecutter.project_name }}/{{ cookiecutter.project_name }}/settings.py]: Yes

File: {{ cookiecutter.project_name }}/{{ cookiecutter.project_name }}/settings_test.py (hunk 2 above)
Comment (noisecapella) [reply {{ cookiecutter.project_name }}/{{ cookiecutter.project_name }}/settings_test.py]: Yes

**Pull request merged**